### DISCUSSION TITLE
Priority on WAN connections HIGH/MID/LOW?

### DISCUSSION THREAD

**Discussion by marccolemont** [Ysurac/openmptcprouter#2701] [Ideas]
I'm looking for a feature that I have available using WMT (Mobile Viewpoint) Video Encoders which also have bonding build-in.
On these units one can set priorities HIGH/MID/LOW on the available connections. For example local venue internet connection is mostly set at HIGH, meaning that it doesn't start consuming data of the SIM cards, unless the data-rate is not sufficient. Having that setting available on any of the connections makes it also possible to set for example an expensive dataprovider at LOW, so that the other SIM cards are being used as much as possible,  unless the data-rate requires more.
It does this feature without any glitch and saved me already a lot of money on SIM card data.
Imagine a 4 day festival livestreaming where it only used a few GB SIM card data over 60 GB of data transferred over the provided internet connection. The SIM cards would run out after a day without this feature.
Is that a feature on the roadmap? If not, this is a request. 
For example:
WAN 1 LAN local internet HIGH
WAN 2 SIM MID
WAN 3 SIM MID
WAN 4 SIM MID
WAN 5 SIM LOW (expensive provider)

**Comment by Ysurac** [comment]
It's on roadmap but not for now. This will need a custom mptcpd path manager.

**Comment by Network-Traditions** [comment]
@Ysurac a similar feature, if not already proposed and/or discussed, would be an ability to schedule WAN usage as well.  For example, Starlink has announced bandwidth limits to be implemented in the near future.  Deprioritized throttling will occur when accounts, subject to the prescribed limit, consume 1 TB of "Peak" usage data within the monthly billing cycle.  There are 2 types of usage: 1) Peak [7AM-11PM] and 2) Off-Peak [11PM-7AM] both based on the subscribers local time zone.  "Off-Peak" data usage remains unlimited.  Subsequently, users are scheduling non-critical data usage (i.e. windows updates, youtube uploads, etc) for "Off-Peak" usage to leverage the upcoming policy change.  Having the ability to schedule the availability of Starlink and other schedule sensitive ISP OMR WAN connected services would be extremely valuable to the project.  Thanks again for your work!
